### PR TITLE
Percent of total in leaderboards

### PR DESCRIPTION
**Pull request by bcolloran**
add percent of total to leaderboards

### CODE REVIEW

**Comment by bcolloran** [comment]
@magorlick and @jkhwu, the leaderboard part of the percent-of-total work is ready for experience review.
A few things to note:
- this is *just* the leaderboard part -- I'll begin working on the dimension table next. My recommendation is that we move ahead with merging this pr before the dimension table work is complete because this adds value and is relatively independent of that.
- this percent-of-total toggle state will not be reflected in the URL. @AdityaHegde, I will need your help with that. My recommendation is that we follow up with a second PR once this is merged; IMO it's better to have this functionality sooner without the URL state than to delay merging in order to finish that detail
- remember that, per the spec, you will need to add `valid_percent_of_total: true` to any measures for which you want to see the percent of total

File: web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte
Comment (AdityaHegde) [inline web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte:110]: Should we move this to a separate component? Showing percents doesnt fit within LeaderboardMeasureSelector.

File: web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte
Comment (AdityaHegde) [inline web-common/src/features/dashboards/leaderboard/LeaderboardMeasureSelector.svelte:36]: nit: remove the log

File: web-common/src/features/dashboards/leaderboard/DimensionLeaderboardEntrySet.svelte
Comment (AdityaHegde) [inline web-common/src/features/dashboards/leaderboard/DimensionLeaderboardEntrySet.svelte:73]: This whole block is rerun for every `comparisonLabelToReveal` change, i.e. hovering over leaderboard values. Perhaps we should have both versions of formatted string and toggle while displaying. I see there is already a FIXME. Any idea on when we are planning to address it?

Comment (bcolloran) [reply web-common/src/features/dashboards/leaderboard/DimensionLeaderboardEntrySet.svelte:73]: I don't know, I was going to mention to @nishantmonu51 that these components have accumulated a great deal of tech debt and badly need some refactoring (I left several other FIXMEs in this PR in addition to this one). It is my recommendation to start on that right away, but it's up to Nishant to make the call.

Comment (bcolloran) [reply web-common/src/features/dashboards/leaderboard/DimensionLeaderboardEntrySet.svelte:73]: @AdityaHegde , discused with Nishant this morning-- the current plan is to tackle these FIXMES after working on the % of total for the dimension table

**Comment by bcolloran** [comment]
Thank you for the helpful review @AdityaHegde! I moved the button toggle to a separate component as you recommended, and renamed `LeaderboardMeasureSelector` to `LeaderboardControls` to clarify that it now does more than just measure selection. This should now be ready for final code review and merge.

@nishantmonu51 and @jkhwu -- if you wish to do a QA/experience pass, this branch is ready (but see notes above)

**Review by AdityaHegde — APPROVED** [review]
Code wise this looks good.

**Comment by nishantmonu51** [comment]
Looks good to be merged to unblock other work.
One boundary case to be fixed here - 

1. Enable `valid_percent_of_total` for metric A
2. Select order by metric A and switch to showing percentage of totals 
3. Switch to a different metric B for which % of total is not enabled
4. Metric B is shown with % of totals 
Expected behavior -> Hide the % of totals column when switching to a metric for which it should not be enabled.

**Review by nishantmonu51 — APPROVED** [review]

**Pull request merged**
